interface/forms/vitals/C_FormVitals.class.php

```diff
@@ -450,9 +450,15 @@ public function populate_object(&$obj)
         foreach ($_POST as $varname => $var) {
             $varname = preg_replace("/[^A-Za-z0-9_]/", "", $varname);
             $func = "set_" . $varname;
+            if ($func == "set_id") {
+                if (!empty($var)) {
+                    $var = (int)$var; // make sure we have an integer
+                } else {
+                    $var = null; // if we don't have an id then we set it to null
+                }
+            }
```

Comment:
I take it 0 is not a valid value for the id?

Reply:
It just seems it might confuse readers, because empty(0) is true, but 0 is a valid integer.